### ISSUE TITLE
Add setting to define app orientation (mainly landscape)

### ISSUE THREAD

**Issue by dennisguse** [EventFahrplan/EventFahrplan#595]
# Preliminaries
Especially for larger events with multiple parallel tracks, the vertical orientation (showing one track only) is not really helpful. 
Horizontal mode is way more helpful as it gives a nice overview what is happening in parallel.

# Current state
Device orientation setting / sensor is used to determine app orientation.

# Target state
Adding a setting to fix the orientation of EventFahrplan to horizontal (overriding device settings).

**Comment by johnjohndoe** [comment]
@dennisguse Thank you for this idea. I actually do not know if that is possible. I never heard of it before. I will check. If you have references on how to implement this feel free to add them here.

**Comment by dennisguse** [comment]
https://developer.android.com/reference/android/app/Activity#setRequestedOrientation(int) 
with 
https://developer.android.com/reference/android/content/pm/ActivityInfo#SCREEN_ORIENTATION_LANDSCAPE

:rocket:

**Comment by johnjohndoe** [comment]
Do you want to implement this as a setting item?

**Comment by dennisguse** [comment]
That was the idea.

**Comment by johnjohndoe** [comment]
Happy New Year! I am looking forward to your contribution. Let me know if you need any help with getting started with the project.

**Comment by dennisguse** [comment]
All good. I have some other problems in OpenTracks I need to fix first ;)
Feature is simple enough :)

**Comment by ldoench** [comment]
I would like to not have a setting to switch the app to landscape mode, but rather select the number of columns.
Should I create a new feature request or can we use this one, since it would also fix the issue behind this one of only showing one track at a time?

Do you have any designs of the app in Penpot or Figma,  so that I can prepare you designs of how a multi column layout in portrait mode could look like?

**Comment by johnjohndoe** [comment]
@ldoench Thank you for request. I am fine with continuing the conversation here or in a new issue.

I have not considered to let users customize the number of columns. The current configuration depends on the screen dimensions (width / height) of the device and the orientation (portrait / landscape).  The `max_cols` resource defines different values for certain configurations, e.g.

- [values/dimens.xml](https://github.com/EventFahrplan/EventFahrplan/blob/47b7cb3fd2883d9e02ab3bdfc95820bd80386760/app/src/main/res/values/dimens.xml#L17): 1
- [values-land/dimens.xml#L21](https://github.com/EventFahrplan/EventFahrplan/blob/47b7cb3fd2883d9e02ab3bdfc95820bd80386760/app/src/main/res/values-land/dimens.xml#L21): 4
- [values-sw600dp-port/dimens.xml](https://github.com/EventFahrplan/EventFahrplan/blob/47b7cb3fd2883d9e02ab3bdfc95820bd80386760/app/src/main/res/values-sw600dp-port/dimens.xml#L18): 4
- [values-sw600dp/dimens.xml#L18](https://github.com/EventFahrplan/EventFahrplan/blob/47b7cb3fd2883d9e02ab3bdfc95820bd80386760/app/src/main/res/values-sw600dp/dimens.xml#L18): 4
- [values-sw720dp/dimens.xml#L18](https://github.com/EventFahrplan/EventFahrplan/blob/47b7cb3fd2883d9e02ab3bdfc95820bd80386760/app/src/main/res/values-sw720dp/dimens.xml#L18): 6

If you want to make your mind up about this I would love to discuss ideas with you. Please don't invest time in coding - we should agree on something before. I do not have design components to be reused right now. I think you can work with your own screenshots to start with.
